File: 5 - Coding Interview Workshops/Week7/CodeFiles/animal.js
class Animal {
    constructor (name) {
        this.name = name;
    }
    
    sayHi () {
        console.log(`Hi, I am a ${this.name}`);
    }
}
class Mammal extends Animal {
    constructor (name, gestationPeriod){
        super(name)
        this.gestation = gestationPeriod
    }
}

let mammal = new Mammal("Zebra", 12);
mammal.sayHi();

File: 5 - Coding Interview Workshops/Week07/CodeFiles/Python/animal.py
class Animal:
    def __init__(self, name):
        self.name = name
    
    def sayHi (self):
        print("Hi, I am a " + self.name)

class Mammal (Animal):
    def __init__(self, name, gestationPeriod):
        super().__init__(name)
        self.gestation = gestationPeriod

mammal = Mammal("Zebra", 12)
mammal.sayHi()

